internal/repository/warmup.go

```diff
@@ -0,0 +1,109 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/restic/restic/internal/backend"
+	"github.com/restic/restic/internal/restic"
+)
+
+type PacksWarmer struct {
+	repo        restic.Repository
+	packs       restic.IDSet
+	packsResult map[restic.ID]error
```

Comment:
I'd prefer to use a sync.Map here. That should clean up the map accesses quite a bit.

Reply:
Well, the sync.Map implementation really looks worse in practice.

- `sync.Map` loses type safety
- because of this, a type assertion is needed in getResult to cast `interface{}` to `error` + a null-assertion in case the error is nil
- a mutex is still used for the initial packs registration. (Well, the mutex could technically be removed because no operation is concurrent in practice, but doing `if !set.Has(x) { set.Insert(x, y) }` without a lock screams future problems)

Reply:
I ended up removing this struct entirely. Resolving.